src/test/java/com/google/sps/ChatServletTest.java

```diff
@@ -0,0 +1,258 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public final class ChatServletTest extends Mockito {
+    private static final String GROUP_ID_PARAM = "groupID";
+    private static final String GROUP_ID = "123";
+    private static final String MESSAGE_TEXT_TOXIC = "what kind of idiot name is foo?";
+    private static final String MESSAGE_KIND = "Message-";
+    private static final String MESSAGE_TEXT_PROPERTY = "message-text";
+    private static final String TIMESTAMP_PROPERTY = "timestamp";
+    private static final String GROUP_KIND = "Group";
+    private static final String GROUP_NAME_PROPERTY = "name";
+    private static final String MESSAGES_MAP_KEY = "messages";
+    private static final String GROUPS_KEY = "groups";
+    private static final ImmutableList<String> SAMPLE_GROUP_LIST = 
+        ImmutableList.of("Black Lives Matter");
+    private static final String CHAT_TEMPLATE = "templates.chat.chatPage";
+    private static final Double SCORE_OFFSET = 0.0000005;
+    private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    private File apiKeyFile;
+    private Scanner scanner;
+    private String apiKey;
+    private String referer;
+    private String apiURL;
+    private StringWriter stringWriter;
+    private PrintWriter printWriter;
+    private SoyFileSet sfs;
+    private SoyTofu tofu;
+    private DatastoreService datastore;
+    private ImmutableMap<String, ImmutableList<String>> templateData;
+
+    private ChatServlet servlet;
+
+    @Mock
+    HttpServletRequest request;
+
+    @Mock
+    HttpServletResponse response;
+
+    @Before
+    public void setUp() {
+        servlet = new ChatServlet();
+        
+        helper.setUp();
+        datastore = DatastoreServiceFactory.getDatastoreService();
+
+        MockitoAnnotations.initMocks(this);
+
+        stringWriter = new StringWriter();
+        printWriter = new PrintWriter(stringWriter);
+
+        sfs = SoyFileSet
+            .builder()
+            .add(new File("src/main/java/templates/chat.soy"))
+            .build();
+        tofu = sfs.compileToTofu();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void testsCommentScore() throws IOException {
+        // Reads API Key and Referer from file 
+        apiKeyFile = new File("src/main/resources/keys.txt");
+        scanner = new Scanner(apiKeyFile);
+        apiKey = scanner.nextLine();
+        referer = scanner.nextLine();
+
+        apiURL = "https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=" + apiKey;
```

Comment:
When you add the String constant in ChatServlet, make sure you update the test to use it too